traffic_ops/testing/api/v4/jobs_test.go

```diff
@@ -360,3 +372,843 @@ func GetTestInvalidationJobs(t *testing.T) {
+	if err != nil {
+		t.Fatalf("unable to get invalidation jobs: %v - alerts: %+v", err, jobs.Alerts)
+	}
+
+	var realJob tc.InvalidationJob
+	for i, job := range jobs.Response {
+		if job.StartTime == nil || job.DeliveryService == nil || job.CreatedBy == nil {
+			t.Error("Traffic Ops returned a representation of a content invalidation Job that had null or undefined Start Time and/or Delivery Service and/or Created By")
+			continue
+		}
+		diff := firstJob.StartTime.Time.Sub(job.StartTime.Time)
+		if *job.DeliveryService == xmlID && *job.CreatedBy == "admin" && diff < time.Second {
+			realJob = jobs.Response[i]
+			break
+		}
+	}
+	if realJob.ID == nil || *realJob.ID == 0 {
+		t.Fatal("could not find new job")
+	}
+
+	//update existing jobs with new ds id
+	originalJob := realJob
+	newTime := tc.Time{
+		Time:  startTime.Time.Add(time.Hour * 2),
+		Valid: true,
+	}
+	originalJob.StartTime = &newTime
+	originalJob.DeliveryService = testData.DeliveryServices[1].XMLID
+	alerts, reqInf, err := TOSession.UpdateInvalidationJob(originalJob, client.RequestOptions{})
+	if err == nil {
+		t.Fatalf("Expected Cannot change 'deliveryService' of existing invalidation job! - alerts: %+v", alerts.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusConflict {
+		t.Errorf("Expected status code 409, got %v", reqInf.StatusCode)
+	}
+
+	//update existing jobs with invalid ds id
+	invaliddsidJob := realJob
+	invaliddsid := "abcd"
+	invaliddsidJob.DeliveryService = &invaliddsid
+	alerts, reqInf, err = TOSession.UpdateInvalidationJob(invaliddsidJob, client.RequestOptions{})
+	if err == nil {
+		t.Fatalf("Expected Cannot change 'deliveryService' of existing invalidation job! - alerts: %+v", alerts.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusConflict {
+		t.Errorf("Expected status code 409, got %v", reqInf.StatusCode)
+	}
+
+	//update existing jobs with blank ds id
+	blankdsidJob := realJob
+	blankdsid := ""
+	blankdsidJob.DeliveryService = &blankdsid
+	alerts, reqInf, err = TOSession.UpdateInvalidationJob(blankdsidJob, client.RequestOptions{})
+	if err == nil {
+		t.Fatalf("Expected deliveryService: cannot be blank. - alerts: %+v", alerts.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//update existing jobs with asset url not starts with origin.infra
+	invalidasseturlJob := realJob
+	assetURL := "http://google.com"
+	invalidasseturlJob.AssetURL = &assetURL
+	alerts, reqInf, err = TOSession.UpdateInvalidationJob(invalidasseturlJob, client.RequestOptions{})
+	if err == nil {
+		t.Fatalf("Expected Cannot set asset URL that does not start with Delivery Service origin URL: http://origin.infra.ciab.test. - alerts: %+v", alerts.Alerts)
+	}
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected status code 400, got %v", reqInf.StatusCode)
+	}
+
+	//update existing jobs with blank asset url
+	blankasseturlJob := realJob
```

Comment:
Same here. This variable should be: blankAssetUrlJob

Reply:
updated.